src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,59 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter
+            .ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        if (dateFrom == null || dateTo == null || names.length == 0 || data.length == 0) {
+            return "Invalid data. Dates should not be null,"
+                    + "also names and data should not be empty";
+        }
+
+        try {
+            LocalDate fromDate = LocalDate.parse(dateFrom, FORMATTER);
+            LocalDate toDate = LocalDate.parse(dateTo, FORMATTER);
+            int[] earnings = new int[names.length];
+            for (String dataUnit : data) {
+                String[] parts = dataUnit.split(" ");
```

Comment:
SEPARATOR

Reply:
not fixed, let's create a constant for this

Reply:
move " " to the constant :)